website/common/script/ops/buy/buyQuest.js

```diff
@@ -49,9 +45,22 @@ export class BuyQuestWithGoldOperation extends AbstractGoldItemOperation {
       throw new NotAuthorized(this.i18n('questNotGoldPurchasable', {key}));
     }
 
+    this.checkPrerequisites(user, key);
+
     this.canUserPurchase(user, item);
   }
 
+  checkPrerequisites (user, questKey) {
+    const item = content.quests[questKey];
```

Comment:
can you pass the item directly instead of passing the key and then refetching the item?

Reply:
I can but I'd have to either also pass the `questKey` param (for the check on line 55) or pull that code back out into a separate method. If I only passed the item, I'd no longer have access to the untranslated quest name.

Which do you prefer... pass item & key, leave it as is, or pull the masterclasser check back out?

Reply:
Ah okay, that makes sense, let's keep it this way